libmachine/mcnutils/b2d.go

```diff
@@ -374,12 +374,17 @@ func (b *B2dUtils) UpdateISOCache(isoURL string) error {
 		}
 	}
 
+	exists := b.exists()
+
 	if isoURL != "" {
+		if exists {
+			// Warn that the b2d iso won't be updated if isoURL is set
+			log.Infof("Boot2Docker URL was explictly set to %s, automatic upgrade is disabled.", isoURL)
```

Comment:
This should probably be set to `log.Warnf` instead and I'd make the copy say: "Boot2Docker URL was explicitly set to %q at create time, so Docker Machine cannot upgrade this machine to the latest version."